### ISSUE TITLE
OpenCL doesn't dump build log when there's a build failure

### ISSUE THREAD

**Issue by pjaaskel** [CHIP-SPV/chip-spv#141]
This is visible e.g. when producing Issue #137.

Level Zero execution prints out a more readable error:
```
234: CHIP error [TID 9571] [1663235346.068551382] : ZE Build Log: 
234: error: Double type is not supported on this platform.
234: in kernel: 'void vectoradd_float<double>(double*, double const*, int, int)'
234: error: backend compiler failed build.
234: 
234: error: Double type is not supported on this platform.
234: in kernel: 'void vectoradd_float<double>(double*, double const*, int, int)'
234: error: backend compiler failed build.
234: 
234: error: Double type is not supported on this platform.
234: in kernel: 'void vectoradd_float<double>(double*, double const*, int, int)'
234: error: backend compiler failed build.
234: 
234: error: Double type is not supported on this platform.
234: in kernel: 'void vectoradd_float<double>(double*, double const*, int, int)'
234: error: backend compiler failed build.
```
while OpenCL just 
```
234: CHIP error [TID 9620] [1663235427.826027232] : hipErrorNotInitialized
234: CHIP error [TID 9620] [1663235427.826069842] : hipErrorLaunchFailure (Failed to find kernel via kernel name: _Z15vectoradd_floatIsEvPT_PKS0_ii) in /home/pjaaskel/src/chip-spv/src/CHIPBackend.cc:242:getKernel
```
making it harder to extract the actual cause.

**Comment by pvelesko** [comment]
Fixed in #130, please check. @pjaaskel

**Comment by pjaaskel** [comment]
Looks good!
